bitcoin/src/address.rs

```diff
@@ -1054,13 +1077,13 @@ impl Address<NetworkUnchecked> {
     /// For details about this mechanism, see section [*Parsing addresses*](Address#parsing-addresses)
     /// on [`Address`].
     #[inline]
-    pub fn assume_checked(self) -> Address { Address::new(self.network, self.payload) }
+    pub fn assume_checked(self) -> Address { Address::new(*self.network(), self.payload().clone()) }
```

Comment:
Clone shouldn't be needed here. If it's an issue, we should have `into_parts()` method. Maybe we should have it anyway.